adjust

```diff
@@ -798,14 +800,66 @@ def test_dep_progress(dep):
     return (progress, "")
 
 
+def compare_settings(patch, dep):
```

Comment:
Consider the following:
* instead of comparing k8s objects, compare the OCO objects instead (the resulting description)
* instead of looking for changes in cpu/mem, look for any changes with the exception of a settings 'replicas'

This should help with the edge case where we have settings that are not resources (i.e. env vars).

Reply:
In general, this is a better approach. It would work nicely if the driver were coded do all updates as a block - and only then wait for them to be applied. However, the patches are applied one component at a time and each one is waited on and checked separately.

Re-working the driver to apply all changes first and then do the waits (in parallel, preferably) is a good thing, but very far from trivial. I have an old note in the code to that effect (line 1139). I don't know if it is worth the effort, given that we are going with ServoX in the future and servo+drivers is a legacy code used only on a limited number of customer deployments. AFAIK, new deployments will be with ServoX exclusively.